File: Ejercicios POO/Python/Herencia y Composicion/Casas.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Written By Ismael Heredia in the year 2016

from Edificios import Edificio
from Cocheras import Cochera

class Casa(Edificio):

   def __init__(self, medidas="",ambientes="",dueno="", telefono="",direccion="",cochera=""):
       Edificio.__init__(self,medidas,ambientes)
       self.__dueno = dueno
       self.__telefono = telefono
       self.__direccion = direccion
       self.__cochera = cochera
              
   def getDueno(self):
       return self.__dueno

   def setDueno(self, dueno = ""):
       self.__dueno = dueno

   def getTelefono(self):
       return self.__telefono

   def setTelefono(self, telefono = ""):
       self.__telefono = telefono
       
   def getDireccion(self):
       return self.__direccion

   def setDireccion(self, direccion = ""):
       self.__direccion = direccion
       
   def getCochera(self):
       return self.__cochera

   def setCochera(self, cochera = ""):
       self.__cochera = cochera
           
   def DatosCasa(self):
       contenido = ""
       contenido += "-- Datos del terreno --\n\n[+] Medidas : %s\n[+] Ambientes : %d\n\n" % (self.getMedidas(),self.getAmbientes())
       contenido += "-- Datos de la casa --\n\n[+] Dueno : %s\n[+] Telefono : %d\n[+] Direccion : %s\n" % (self.__dueno,self.__telefono,self.__direccion)
       if self.__cochera!=None:
           contenido += "[+] Cochera : %s" % (self.__cochera.getDimensiones())
       return contenido
		
   def DatosCasaCompleto(self):
       contenido = ""
       contenido += "-- Datos del terreno --\n\n[+] Medidas : %s\n[+] Ambientes : %d\n\n" % (self.getMedidas(),self.getAmbientes())
       contenido += "-- Datos de la casa --\n\n[+] Dueno : %s\n[+] Telefono : %d\n[+] Direccion : %s\n" % (self.__dueno,self.__telefono,self.__direccion)
       if self.__cochera!=None:
           contenido += "\n-- Datos de la cochera --\n\n[+] Ambientes : %d\n[+] Dimensiones : %s\n" % (self.__cochera.getAmbientes(),self.__cochera.getDimensiones())
           if self.__cochera.getAuto()!=None:
               contenido += "\n-- Datos del auto --\n\n[+] Marca : %s\n[+] Numero serie : %s" % (self.__cochera.getAuto().getMarca(),self.__cochera.getAuto().getNumero_serie())
       return contenido
          
   def destroy(self):
       self.__dueno = None
       self.__telefono = None
       self.__direccion = None
       self.__cochera = None

File: Ejercicios POO/PHP/Herencia y Composicion/Casas.php
<?php

// Written By Ismael Heredia in the year 2016

class Casa extends Edificio
{
    
    private $dueño;
    private $telefono;
	private $direccion;
	private $cochera;
    
    public function __construct()
    {
        $this->dueño   = "";
        $this->telefono = "";
		$this->direccion = "";
		$this->cochera = "";
    }
    
    public static function CreateCasa($medidas,$ambientes,$dueño,$telefono,$direccion,$cochera)
    {
        $instance                 = new self();
		$instance->medidas   = $medidas;
		$instance->ambientes   = $ambientes;
        $instance->dueño   = $dueño;
        $instance->telefono = $telefono;
		$instance->direccion = $direccion;
		$instance->cochera = $cochera;
        return $instance;
    }
    
    public function getDueño()
    {
        return $this->dueño;
    }
    
    public function setDueño($dueño)
    {
        $this->dueño = $dueño;
    }
    
    public function getTelefono()
    {
        return $this->telefono;
    }
    
    public function setTelefono($telefono)
    {
        $this->telefono = $telefono;
    }
	
    public function getDireccion()
    {
        return $this->direccion;
    }
    
    public function setDireccion($direccion)
    {
        $this->direccion = $direccion;
    }
	
    public function getCochera()
    {
        return $this->cochera;
    }
    
    public function setCochera($cochera)
    {
        $this->cochera = $cochera;
    }
        
    public function __destruct()
    {
        $this->dueño   = "";
        $this->telefono = "";
		$this->direccion = "";
		$this->cochera = "";
    }
	
    public function DatosCasa()
    {
    	$contenido = "";
		
        $contenido = $contenido . "-- Datos del terreno --" . "\n\n" .
                "[+] Medidas : " . $this->medidas . "\n" .
                "[+] Ambientes : " . $this->ambientes . "\n\n";

         $contenido = $contenido . "-- Datos de la casa --" . "\n\n"
            . "[+] Dueño : " . $this->dueño . "\n"
            . "[+] Telefono : " . $this->telefono . "\n"
            . "[+] Direccion : " . $this->direccion . "\n";
		
		if(!is_null($this->cochera)) {	
    		$contenido =  $contenido . "[+] Cochera : " . $this->cochera->getDimensiones() . "\n";
		}
					
		return $contenido;
    }
    
    public function DatosCasaCompleto()
    {
    	$contenido = "";
		
        $contenido = $contenido . "-- Datos del terreno --" . "\n\n" .
                "[+] Medidas : " . $this->medidas . "\n" .
                "[+] Ambientes : " . $this->ambientes . "\n\n";

         $contenido = $contenido . "-- Datos de la casa --" . "\n\n"
            . "[+] Dueño : " . $this->dueño . "\n"
            . "[+] Telefono : " . $this->telefono . "\n"
            . "[+] Direccion : " . $this->direccion . "\n";
		
		if(!is_null($this->cochera)) {
			
    		$contenido =  $contenido . "\n-- Datos de la cochera -- " . "\n\n"
                . "[+] Ambientes : " . $this->cochera->getAmbientes() . "\n"
                . "[+] Dimensiones : " . $this->cochera->getDimensiones() . "\n";
		
				
			if(!is_null($this->cochera->getAuto())) {
        		$contenido = $contenido . "\n-- Datos del auto --" . "\n\n"
                . "[+] Marca : " . $this->cochera->getAuto()->getMarca() . "\n"
                . "[+] Numero de serie : " . $this->cochera->getAuto()->getNumero_serie() . "\n";	
			}
		
		}
			
		return $contenido;
    }
    
}

?>